client/web/src/repo/blob/codemirror/search.tsx

```diff
@@ -243,21 +324,54 @@ class SearchPanel implements Panel {
                     }
                 }
 
-                const matchLineBlock = this.view.lineBlockAt(result.value.from)
-                const matchLineCenter = matchLineBlock.top + matchLineBlock.height / 2
+                this.updateSelectedSearchMatch(result.value)
 
-                if (matchLineCenter < scrollTop || matchLineCenter > scrollTop + this.view.scrollDOM.clientHeight) {
-                    this.view.dispatch({
-                        effects: EditorView.scrollIntoView(result.value.from, {
-                            y: 'center',
-                        }),
-                    })
-                }
+                // Taken from the original `findPrevious` and `findNext` CodeMirror implementation:
+                // https://github.com/codemirror/search/blob/affb772655bab706e08f99bd50a0717bfae795f5/src/search.ts#L385-L416
+                this.view.dispatch({
+                    selection: { anchor: result.value.from, head: result.value.to },
+                    scrollIntoView: true,
+                    effects: announceMatch(this.view, result.value),
+                    userEvent: 'select.search',
+                })
             }
         }
     }
 }
 
+// Announce the current match to screen readers.
+// Taken from original the CodeMirror implementation:
+// https://github.com/codemirror/search/blob/affb772655bab706e08f99bd50a0717bfae795f5/src/search.ts#L694-L717
+const AnnounceMargin = 30
+const Break = /[\s!,.:;?]/
```

Comment:
Should we lowercase these variables so they are more in line with our code style?

```suggestion
const announceMargin = 30
const break = /[\s!,.:;?]/
```

Reply:
Addressed in https://github.com/sourcegraph/sourcegraph/pull/45416/commits/28ceb5c57ae1b2e1be91d6a4fa7a675a9b6519a9.